src/logdb/test/logdb_tests.c

```diff
@@ -120,7 +120,7 @@ void test_logdb(logdb_log_db* (*new_func)())
     u_assert_int_eq(memcmp(value_test->str, value->str, value->len), 0);
 
     value_test = logdb_find(db, key2);
-    u_assert_int_eq((int)value_test, 0); /* should be null */
+    u_assert_uint32_eq((uintptr_t)value_test, 0); /* should be null */
```

Comment:
But pointers have a size of 64bit on 64bit CPUs, and 32bit on 32bit CPUs.  
So on a 64bit machine will still produce a warning.  
The warning is not completely harmless in this case due to possible overflow: could cause the test to pass even with a non-null pointer.  
If you use uint64_t, it should be able to contain both 32bit and 64bit pointers without any overflow, but you still get the warning due to being of different sizes.
The type uintptr_t is part of the C99 standard. It will be equivalent to uint32_t on a 32bit machine, and uint64_t on a 64bit machine, so should be perfect.
See http://stackoverflow.com/questions/1845482/what-is-uintptr-t-data-type
Follows C99 standard: http://en.cppreference.com/w/c/types/integer